From 06f1d0a08dadea976a2ed4a57b3887e09599c600 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 28 Nov 2022 10:28:07 +0200
Subject: [PATCH] Revert "[SW-74767] gaudi3: save ppboot version to bootinfo"

This reverts commit 46343a6c1aa1f79c369c57293597994ca49cbf05.

this commit brakes the simulator run, and also calling
boot_info_finalize_no_merge() at this phase is wrong, because
the bootinfo data was set by bootrom, and should be used by later
components.

Change-Id: Iacc36b01d508e55c4571fdeb3c16d6c4454c3e7f
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 3f62e54..bf2cc26 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -238,12 +238,10 @@
 __weak void ppboot_pcie_drain_init(void) {}
 
 __weak void hl_plat_init(void) {}
-__weak void ppboot_send_version(void) {}
 
 static inline void ppboot_version(void)
 {
 	preboot_puts("Ppboot " VERSION_STR "\n");
-	ppboot_send_version();
 }
 
 __weak void hl_enter_recovery(void)
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 458b23c..bdd9fae 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -895,13 +895,3 @@
 	if (err)
 		preboot_puts_hex("failed to copy phy fw to pcie sram, err 0x", err, false);
 }
-
-void ppboot_send_version(void)
-{
-	uint32_t ver_size = sizeof(PREBOOT_VER);
-
-	if (boot_info_set_no_merge(HL_BOOT_INFO_PPBOOT_VERSION, (void *)PREBOOT_VER, ver_size))
-		preboot_puts("BOOT_INFO: Failed to set ppboot version\n");
-
-	boot_info_finalize_no_merge();
-}
diff --git a/include/preboot.h b/include/preboot.h
index 97c83ef..54086c7 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -262,7 +262,6 @@
 void hl_flr_report_err(u32 err_bit);
 void preboot_security_sp_sram_init(void);
 u32 hl_get_pll_div_fix(void);
-void ppboot_send_version(void);
 
 #endif /* __ASSEMBLY__ */
 
